File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    memo = [0 for _ in range(n+1)]
    (memo[0], memo[1]) = (0, 1)
    
    for i in range(2, n+1):
        memo[i] = (memo[i-1] + memo[i-2]) % 1234567
    
    return memo[n]

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.swift
func solution(_ n:Int) -> Int {
    var table = Array(repeating: 0, count: n + 1)
    table[1] = 1
    
    for i in 2...n {
        table[i] = (table[i - 1] + table[i - 2]) % 1234567
    }
    
    return table[n]
}